default_validator/src/lib.rs

```diff
@@ -0,0 +1,132 @@
+        $crate::isi::trigger::map_tokens!($callback);
+        $crate::isi::validator::map_tokens!($callback);
+    };
+}
+
+pub(crate) use map_all_crate_tokens;
+
+/// Validate operation.
+#[entrypoint(params = "[authority, operation]")]
+pub fn validate(
+    authority: <Account as Identifiable>::Id,
+    operation: NeedsPermissionBox,
+) -> Verdict {
+    match operation {
+        NeedsPermissionBox::Transaction(transaction) => validate_and_execute_transaction(
+            &authority,
+            &transaction,
+            validate_and_execute_instruction,
+            validate_query,
+        ),
+        NeedsPermissionBox::Instruction(instruction) => {
+            validate_and_execute_instruction(&authority, &instruction, validate_query)
+        }
+        NeedsPermissionBox::Query(query) => validate_query(&authority, query),
+    }
+}
+
+/// Default validation of [`SignedTransaction`].
+///
+/// Performs execution if transaction contains [`Executable::Instructions`]
+/// and does nothing if [`Executable::Wasm`].
+///
+/// Execution is done to properly validate dependent instructions.
+pub fn validate_and_execute_transaction<I, Q>(
+    authority: &<Account as Identifiable>::Id,
+    transaction: &SignedTransaction,
+    validate_and_execute_instruction: I,
+    validate_query: Q,
+) -> Verdict
+where
+    I: Fn(&<Account as Identifiable>::Id, &InstructionBox, Q) -> Verdict + Copy,
+    Q: Fn(&<Account as Identifiable>::Id, QueryBox) -> Verdict + Copy,
+{
+    match transaction.payload().instructions() {
+        Executable::Wasm(_) => {
+            // WASM validation is done by execution on Iroha side.
+            // All instructions executed by WASM will be passed as
+            // [`NeedsPermissionBox::Instruction`] to this validator.
+            pass!()
+        }
+        Executable::Instructions(instructions) => {
+            for isi in instructions {
+                validate_and_execute_instruction(authority, isi, validate_query);
+            }
+            pass!()
+        }
+    }
+}
+
+/// Default validation of [`Instruction`].
+/// Fallbacks to [`DefaultValidate`] and execute instruction on success.
+///
+/// Execution is done to properly validate dependent instructions.
+///
+/// `validate_query` function used to validate queries inside instruction expressions.
+pub fn validate_and_execute_instruction<Q>(
+    authority: &<Account as Identifiable>::Id,
+    instruction: &InstructionBox,
+    validate_query: Q,
+) -> Verdict
+where
+    Q: Fn(&<Account as Identifiable>::Id, QueryBox) -> Verdict + Copy,
+{
+    let verdict = instruction.default_validate(authority, validate_query);
+    if verdict.is_pass() {
+        instruction.execute()
+    }
+    verdict
+}
+
+/// Default validation for [`QueryBox`].
+/// Always returns [`Pass`](Verdict::Pass).
+#[allow(clippy::needless_pass_by_value)]
+pub fn validate_query(_authority: &<Account as Identifiable>::Id, _query: QueryBox) -> Verdict {
```

Comment:
I don't think so. Default configuration is for public blockchain. And there is no meaning in queires permissions in a public blockchain, because anyone can be a node